src/main/scala/castalia/matcher/MatchResultGatherer.scala

```diff
@@ -0,0 +1,62 @@
+package castalia.matcher
+
+import akka.actor._
+import akka.http.scaladsl.model.StatusCodes._
+import akka.pattern.{ask, pipe}
+import akka.util.Timeout
+import castalia.matcher.MatchResultGatherer.{MatchFound, MatchNotFound}
+import castalia.model.Model.StubResponse
+
+import scala.concurrent.ExecutionContext.Implicits.global
+import scala.concurrent.duration._
+
+/**
+  * Created by Jordi on 4-2-2016.
+  */
+object MatchResultGatherer {
+
+  case object MatchNotFound
+
+  case class MatchFound(stubHandler: ActorRef, requestMatch: RequestMatch)
+
+  def props(nOfMatchers: Int, origin: ActorRef): Props = Props(new MatchResultGatherer(nOfMatchers, origin))
+}
+
+class MatchResultGatherer(nrOfMatchers: Int, origin: ActorRef) extends Actor with ActorLogging {
+  implicit val timeout = Timeout(2.seconds)
+
+  def receive: Receive = awaitResponses(false, nrOfMatchers)
+
+  def awaitResponses(responseSent: Boolean, responsesToGet: Int): Receive = {
+    case MatchNotFound =>
+      handleMatchNotFound(responseSent, responsesToGet)
+    case MatchFound(handler, requestMatch) =>
+      handleMatchFound(responseSent, responsesToGet, handler, requestMatch)
+  }
+
+  def handleMatchNotFound(responseSent: Boolean, responsesToGet: Int): Unit = {
+    if (responsesToGet > 1) {
```

Comment:
![Codacy](https://www.codacy.com/versioned/images/favicon.png) Issue found: [Consider using case matching instead of else if blocks](https://www.codacy.com/app/scala-academy/Castalia/file/1993187305/issues/source?bid=2998213&fileBranchId=3120054#l38)